pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -839,35 +857,43 @@ func TestEventualConsistency(t *testing.T) {
 
 		// First we simulate NodeUpdate that should delete 'pod1'. It doesn't know about 'pod2' yet.
 		controller.NodeUpdated(item.oldNode, item.newNode)
-		// TODO(mborsz): Remove this sleep and other sleeps in this file.
-		time.Sleep(timeForControllerToProgress)
+
+		actions := fakeClientset.Actions()
+		if item.expectDelete {
+			actions = getDeleteActions(fakeClientset)
+		}
 
 		podDeleted := false
-		for _, action := range fakeClientset.Actions() {
+		for _, action := range actions {
 			if action.GetVerb() == "delete" && action.GetResource().Resource == "pods" {
 				podDeleted = true
 			}
 		}
-		if !podDeleted {
-			t.Errorf("%v: Unexpected test result. Expected delete, got: %v", item.description, podDeleted)
+
+		if podDeleted != item.expectDelete {
+			t.Errorf("%v: Unexpected test result. Expected delete %v, got %v", item.description, item.expectDelete, podDeleted)
 		}
+
 		fakeClientset.ClearActions()
 
-		// And now the delayed update of 'pod2' comes to the TaintManager. We should delete it as well.
 		holder.setPod(item.newPod)
 		controller.PodUpdated(item.prevPod, item.newPod)
-		// wait a bit
-		time.Sleep(timeForControllerToProgress)
+
+		actions = fakeClientset.Actions()
+		if item.expectDelete {
+			actions = getDeleteActions(fakeClientset)
+		}
 
 		podDeleted = false
-		for _, action := range fakeClientset.Actions() {
+		for _, action := range actions {
 			if action.GetVerb() == "delete" && action.GetResource().Resource == "pods" {
 				podDeleted = true
 			}
 		}
+
 		if podDeleted != item.expectDelete {
 			t.Errorf("%v: Unexpected test result. Expected delete %v, got %v", item.description, item.expectDelete, podDeleted)
 		}
-		close(stopCh)
+		cancel()
```

Comment:
Call cancel in defer. This will make sure it is actually called.

Reply:
Fixed